src/idpi/operators/curl.py

```diff
@@ -27,8 +27,8 @@ def curl(
     tgrlat = cast(xr.DataArray, np.tan(rlat))
 
     # compute weighted derivatives for FD
-    u_f = destagger(u, "x")
-    v_f = destagger(v, "y")
+    u_f = destagger(u, "x") if u.origin_x != 0.0 else u
```

Comment:
I suppose it could be applied in those cases but at that point we should maybe consider making the destagger operator a no op when the field is not staggered. I'm just not sure if this is something that users would expect.

Reply:
https://meteoswiss.atlassian.net/browse/APNRZ-382